components/proxy-service/before-commit.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+readonly CI_FLAG=ci
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+INVERTED='\033[7m'
+NC='\033[0m' # No Color
+
+echo -e "${INVERTED}"
+echo "USER: " + $USER
+echo "PATH: " + $PATH
+echo "GOPATH:" + $GOPATH
+echo -e "${NC}"
+
+##
+# DEP ENSURE
+##
+dep ensure -v --vendor-only
+ensureResult=$?
+if [ ${ensureResult} != 0 ]; then
+	echo -e "${RED}✗ dep ensure -v --vendor-only${NC}\n$ensureResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ dep ensure -v --vendor-only${NC}"
+fi
+
+##
+# GO BUILD
+##
+buildEnv=""
+if [ "$1" == "$CI_FLAG" ]; then
+	# build binary statically
+	buildEnv="env CGO_ENABLED=0"
+fi
+
+${buildEnv} go build -o proxyservice ./cmd/proxyservice
+
+goBuildResult=$?
+if [ ${goBuildResult} != 0 ]; then
+	echo -e "${RED}✗ go build${NC}\n$goBuildResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ go build${NC}"
+fi
+
+##
+# DEP STATUS
+##
+echo "? dep status"
```

Comment:
`dep status` will fail on CI as Gopkg.lock was generated by `dep` with version higher than 0.4.1, which is the version used by buildpack.